RecoTracker/TkSeedGenerator/plugins/DeepCoreSeedGenerator.h

```diff
@@ -0,0 +1,177 @@
+#define Nlayer 4    //Number of layer used in DeepCore
+#define Nover 3     //Max number of tracks recorded per pixel
+#define Npar 5      //Number of track parameter
+
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "Geometry/CommonDetUnit/interface/GlobalTrackingGeometry.h"
+#include "Geometry/Records/interface/GlobalTrackingGeometryRecord.h"
+#include "Geometry/CommonDetUnit/interface/GlobalTrackingGeometry.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSet.h"
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/JetReco/interface/Jet.h"
+#include "DataFormats/SiPixelDigi/interface/PixelDigi.h"
+#include "DataFormats/GeometryVector/interface/VectorUtil.h"
+#include "DataFormats/SiPixelDetId/interface/PXBDetId.h"
+#include "DataFormats/Math/interface/Point3D.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
+
+#include "RecoLocalTracker/ClusterParameterEstimator/interface/PixelClusterParameterEstimator.h"
+#include "RecoLocalTracker/Records/interface/TkPixelCPERecord.h"
+
+#include "SimDataFormats/TrackerDigiSimLink/interface/PixelDigiSimLink.h"
+
+#include "TrackingTools/GeomPropagators/interface/StraightLinePlaneCrossing.h"
+#include "TrackingTools/GeomPropagators/interface/Propagator.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+
+#include <boost/range.hpp>
+#include <boost/foreach.hpp>
+#include "boost/multi_array.hpp"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "SimDataFormats/Vertex/interface/SimVertex.h"
+
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+
+#include "TTree.h"
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+namespace edm {
+  class Event;
+  class EventSetup;
+}  // namespace edm
+
+class DeepCoreSeedGenerator : public edm::one::EDProducer<edm::one::SharedResources> {
```

Comment:
merge .h into .cc

Why is it an edm::one?
We need a very good justification for it. If it has to be edm::one, all CPU cost estimates will have to be scaled up by at least 8 if done on single-thread setup

Reply:
There is no strong justification for that, it can be removed. What do you suggest to use instead?

Reply:
ideally, `edm::global`, but given the use of TF for which we do not yet have examples with reentrant use, the practical choice is `edm::stream`

Reply:
ok I moved to edm::stream, thank you for the suggestion.

Reply:
I noted only now, I'm sorry, what do you mean for "merge .h into .cc"?

Reply:
> what do you mean for "merge .h into .cc"?

to be more explicit:
copy contents of .h to .cc, then remove.h file

Reply:
ok, I'll do that (next commit), but can I ask you why? (I was following the approach of the other plugins of RecoTracker/TkSeedgenerator)